File: ADMIN/Example-API-Flutter/Example.UserService.WebApp/Controllers/HomeController.cs
using Example.UserService.WebApp.Authorization;
using AuthCustom = Example.UserService.WebApp.Authorization.AuthorizeAttribute;
using Example.UserService.WebApp.Const;
using Example.UserService.WebApp.Helpers;
using Example.UserService.WebApp.Models;
using Example.UserService.WebApp.Services.IServices;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;

namespace Example.UserService.WebApp.Controllers
{
    [AuthCustom]
    public class HomeController : Controller
    {
        private readonly IHomeService _service;

        public HomeController(IHomeService service) => _service = service;

        public async Task<IActionResult> Index()
        {
            ViewData["Title"] = "Trang chủ";
            ViewData["ActiveMenu"] = "Home";

            var userId = SessionHelper.GetUserId(HttpContext);
            var vm = await _service.BuildDashboardAsync(
                userId,
                HttpContext.Session.GetString(SessionKeys.FullName),
                HttpContext.Session.GetString(SessionKeys.Username));

            return View(vm);
        }

        [AllowAnonymous]
        public IActionResult Error() => View(new ErrorViewModel { RequestId = HttpContext.TraceIdentifier });
    }
}

File: lib/controllers/home_controller.dart
import 'package:flutter/foundation.dart';
import '../services/home_service.dart';

/// Controller trang chủ — tương đương `HomeController` (WebApp).
class HomeController extends ChangeNotifier {
  HomeController(this._service);

  final HomeService _service;

  bool _isLoading = false;
  DashboardData? _dashboard;

  bool get isLoading => _isLoading;
  DashboardData? get dashboard => _dashboard;

  Future<void> loadDashboard({
    required int userId,
    String? fullName,
    String? username,
  }) async {
    _isLoading = true;
    notifyListeners();

    _dashboard = await _service.buildDashboard(
      userId: userId,
      fullName: fullName,
      username: username,
    );

    _isLoading = false;
    notifyListeners();
  }
}
